services/web/server/src/simcore_service_webserver/session/_cookie_storage.py

```diff
@@ -0,0 +1,72 @@
+"""
+Extends aiohttp_session.cookie_storage
+
+"""
+
+import logging
+import time
+
+import aiohttp_session
+from aiohttp import web
+from aiohttp_session.cookie_storage import EncryptedCookieStorage
+
+_logger = logging.getLogger(__name__)
+
+
+def _share_cookie_across_all_subdomains(
+    request: web.BaseRequest, params: aiohttp_session._CookieParams
+) -> aiohttp_session._CookieParams:
+    """
+    Shares cookie across all subdomains, by appending a dot (`.`) in front of the domain name
+    overwrite domain from `None` (browser sets `example.com`) to `.example.com`
+    """
+    if (host := request.url.host) and host is not None:
+        params["domain"] = f".{host.lstrip('.')}"
```

Comment:
This has to fail on the host.lstrip in production if the code is None. You are not allowed to optionally set the domain you either set it or raise and error. If you leave the domain as unset, or None you are breaking authentication on all subdomains. Please remove the if and use the previous implementation